modules/terraform/format.go

```diff
@@ -140,21 +140,22 @@ func mapToHclString(m map[string]interface{}) string {
 
 // Convert a primitive, such as a bool, int, or string, to an HCL string. If this isn't a primitive, force its value
 // using Sprintf. See ToHclString for details.
-func primitiveToHclString(value interface{}) string {
+func primitiveToHclString(value interface{}, isChild bool) string {
 	switch v := value.(type) {
 
-	// Terraform treats a boolean true as a 1 and a boolean false as a 0. It's best to convert to these ints when
-	// passing booleans as -var parameters. Moreover, due to a Terraform bug
-	// (https://github.com/hashicorp/terraform/issues/7962), all ints must be wrapped as strings.
 	case bool:
 		if v {
 			return "1"
 		}
 		return "0"
 
-	// Note: due to a Terraform bug (https://github.com/hashicorp/terraform/issues/7962), we can't use proper HCL
-	// syntax for ints have to wrap them as strings by falling through to the default case
-	//case int: return strconv.Itoa(v)
+	case string:
+		// If string is a child of a larger data structure (e.g. list of string, map of string), ensure value is quoted
+		if isChild {
+			return fmt.Sprintf("\"%v\"", v)
```

Comment:
Would it make more sense to do the inverse? That is, wrap _all_ strings, but then in the top-level method, if the value is returned, _unwrap_ the quotes there? That way, you don't have to pass this weird boolean all the way through.

Reply:
IMO, I think this is actually easier to understand from an algorithm perspective. It is much easier to understand mentally in the context of `primitiveToHclString` what to do when nested in a larger structure, than it is to think about what we need to do about what `primitiveToHclString` returns in the context of the top most level, worrying about whether the nested value was a string or bool or int (which we don't know at the top level).

Reply:
We could probably rename the parameter to `isNested` to make it clearer

Reply:
I like that name better, so went ahead and did that.